workflow/config/config.go

```diff
@@ -83,6 +83,8 @@ type KubeConfig struct {
 type ArtifactRepository struct {
 	// ArchiveLogs enables log archiving
 	ArchiveLogs *bool `json:"archiveLogs,omitempty"`
+	// AutoArchiveLogs enables log archiving for all tasks regardless of workflow configuration
+	AutoArchiveLogs *bool `json:"autoArchiveLogs,omitempty"`
```

Comment:
Maybe `ForceArchiveLogs` is more explicit?

Reply:
How about `DefaultArchiveLogs`? This change wouldn't _force_ logs to be archived (users explicitly setting s3/artifactory destinations in template wouldn't be covered by this).